### PR TITLE
A fresh look for  our dox

### PR DESCRIPTION
**Pull request by gheber**
Take inspiration from Eigen3
- Updated HTML variables
- HTML header and footer + stylesheet
- Enabled local search

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1117,17 +1117,27 @@ if test "X$HDF5_DOXYGEN" = "Xyes"; then
...
   DOXYGEN_INPUT_DIRECTORY='$(SRCDIR) $(SRCDIR)/doxygen/dox'
   DOXYGEN_OPTIMIZE_OUTPUT_FOR_C=YES
   DOXYGEN_MACRO_EXPANSION=YES
   DOXYGEN_OUTPUT_DIRECTORY=hdf5lib_docs
   DOXYGEN_EXAMPLES_DIRECTORY='$(SRCDIR)/doxygen/examples'
+  DOXYGEN_LAYOUT_FILE=hdf5doxy_layout.xml
+  DOXYGEN_HTML_HEADER=hdf5_header.html
+  DOXYGEN_HTML_FOOTER=hdf5_footer.html
+  DOXYGEN_HTML_EXTRA_STYLESHEET=hdf5doxy.css
+  DOXYGEN_HTML_EXTRA_FILES='hdf5_navtree_hacks.js img/ftv2node.png img/ftv2pnode.png'
```
Comment (ChristopherHogan) [inline configure.ac]: Should these use absolute paths with `SRCDIR` to support out-of-tree builds? When I run the standard

```bash
mkdir build
cd build
/path/to/hdf5/configure --enable-doxygen
make doxygen
```
Doxygen can't find these files.

File: doxygen/dox/Overview.dox [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+<a href="hdf5-doc.tgz">download</a> it as a tgz archive for offline reading.
+
+This is the documention set for HDF5 in terms of specifications and software
+developed and maintained by <a href="https://www.hdfgroup.org/">The HDF
+Group</a>. It is impractical to document the entire HDF5 ecosystem in one place,
+and you should also consult the documentation sets of the many outstanding
+community projects.
+
+For a first contact with HDF5, the best place is to have a look at the \link
+GettingStarted getting started \endlink page that show you how to write and
```
Comment (ChristopherHogan) [inline doxygen/dox/Overview.dox]: "Show" should be "shows."

File: doxygen/dox/ReferenceManual.dox [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/** \page RM Reference Manual
+
+The HDF5 C-API provides applications with fine-grained control over all
+aspects HDF5 functionality. This functionality is grouped into the following
```
Comment (ChristopherHogan) [inline doxygen/dox/ReferenceManual.dox]: "aspects **of**"

**Comment by ChristopherHogan** [comment]
There are several references to "Eigen" in comments. I'm not sure if you want to leave those for attribution, or change them to "HDF5."

File: configure.ac (hunk 1 above)
Comment (gheber) [reply configure.ac]: Good point. Let me fix this.

**Review by ChristopherHogan — APPROVED** [review]

**Comment by epourmal** [comment]
Looks good to me.

**Review by epourmal — APPROVED** [review]

**Pull request merged**